[fix] 변경된 디자인에 맞게 뷰 기능 변경

kelly-chui opened the pull request:
### 🔖  Issue Number

close #190 

### 📙 작업 내역

> 구현 내용 및 작업 했던 내역을 적어주세요.
> 
- [X]  홈 뷰에서 메이트 리스트 뷰로 가는 것으로 버튼 기능 변경
- [X]  메이트 리스트 뷰에서 NI, MPC 세션 활성화 하는 것으로 버튼 기능 변경
- [X]  버튼 config 변화 액션 추가

https://github.com/user-attachments/assets/ee5dabf6-0810-4540-bc0a-571c7ed528af

### 📝 PR 특이사항 

> PR을 볼 때 팀원에게 알려야 할 특이사항을 알려주세요.
> 
- 상황에 따라 버튼의 상태가 변화합니다.
- normal, connectiong, success, failure와 같은 상태가 있습니다.

Commit:
[delete] #190 탭 바 walk 아이템 비활성화

- 탭 바의 walk 아이템을 비활성화 했습니다.

Commit:
[fix] #190 홈 모듈 새로운 디자인에 맞게 정리

- 기존에 있던 NI, MPC 관련 코드를 모두 제거했습니다.
- 산책 신청하기 버튼을 누르면 메이트 목록 뷰로 이동하도록 했습니다.
- 메이트 목록 뷰로 이동할 때는 탭 바 컨트롤러의 인덱스를 변경하는 방식으로 이동합니다.

Commit:
[feat] 메이트 목록 뷰에 MPC, NI 기능 추가

- 원래 홈 뷰에 있던 방식으로 추가했습니다.

Commit:
[chore] 프로젝트 파일 업데이트

pearhyunjin committed:
[fix] #190 바인딩 문제 수정

Commit:
[fix] #190 홈 라우터 메이트 신청 뷰 라우팅 제거

Commit:
[feat] #190 버튼 config 변경 기능 추가

- 현재 NI, MPC의 상태에 따라서 버튼의 config이 변화하도록 했습니다.

Commit:
[fix] #190 실패 했을때 버튼 상태 되돌리기

Commit:
[feat] #190 버튼 상태 실패 추가, 버튼 상태 변경 세분화

- 버튼 상태 변경을 세분화했습니다. 실패 한 경우에도 버튼의 상태가 변화합니다.

Commit:
[fix] #190 메이트 추가 버튼 기본 색상, 이미지 변경

pearhyunjin reviewed (approved):
고생하셨습니다! 잘 변경된 것 같아요~
버튼에 연결 상태가 표시되도록 변경했으니, alert을 좀 변경해봐도 좋을 것 같다는 생각이 드네요

green-yoon87 reviewed (approved):
오! 디자인 바뀐거 적용 잘 하셨네요! 수고하셨습니ㅏ~

Commit:
[fix] #190 버튼 상태 변경 시점 변경

- 버튼의 성공/실패 시점이 NI를 기준으로 해야합니다.
- 현재 NI Manager가 성공/실패를 알리는 동작이 없으므로 테이블 셀의 리로드 데이터를 기준으로 버튼의 상태를 변화시킵니다.

kelly-chui committed:
Merge branch 'dev' into feature/#190/home-view-mate-list-view-design-change
